pkg/kubelet/kubelet_pods.go

```diff
@@ -1107,11 +1108,18 @@ func (kl *Kubelet) HandlePodCleanups() error {
 		klog.ErrorS(err, "Error listing containers")
 		return err
 	}
+
+	var wg sync.WaitGroup
 	for _, pod := range runningPods {
 		if _, found := desiredPods[pod.ID]; !found {
-			kl.podKiller.KillPod(&kubecontainer.PodPair{APIPod: nil, RunningPod: pod})
+			wg.Add(1)
+			go func(pod *container.Pod) {
+				defer wg.Done()
+				<-kl.podKiller.KillPod(&kubecontainer.PodPair{APIPod: nil, RunningPod: pod})
```

Comment:
Can we swap both lines to avoid the `defer`?

Reply:
fixed!